<launch>  

  <node name="video"    pkg="usb_cam"    type="usb_cam_node" respawn="true">
    <param name="video_device" value="/dev/video0" />
    <param name="pixel_format" value="yuyv" />
  </node>

  <group ns="color_filter">
    <node name="filter" pkg="color_filter" type="color-filter-hsv-node">
      <remap from="in" to="/video/image_raw" />
    </node>
  </group>
  <node name="filtered_view" pkg="image_view" type="image_view">
    <remap from="image" to="color_filter/filtered"/>
  </node>


  <group ns="focus">
    <node name="locus" pkg="focus" type="locus.py">
      <remap from="in/compressed" to="/color_filter/filtered/compressed"/>
    </node>
  </group>


  <node name="focus_view" pkg="image_view" type="image_view">
    <param name="image_transport" value="compressed"/>
    <remap from="image" to="/focus/out"/>
  </node>

  <node name="reconf" pkg="rqt_reconfigure" type="rqt_reconfigure"/>
  
</launch>
